solr/core/src/java/org/apache/solr/handler/api/V2ApiUtils.java

```diff
@@ -87,13 +88,14 @@ public static void squashIntoNamedListWithoutHeader(
     squashObjectIntoNamedList(destination, toSquash, true);
   }
 
-  public static String getMediaTypeFromWtParam(
-      SolrQueryRequest solrQueryRequest, String defaultMediaType) {
-    final String wtParam = solrQueryRequest.getParams().get(WT);
-    if (wtParam == null) return "application/json";
+  public static String getMediaTypeFromWtParam(SolrParams params, String defaultMediaType) {
+    final String wtParam = params.get(WT);
+    if (StrUtils.isBlank(wtParam)) return defaultMediaType;
 
     // The only currently-supported response-formats for JAX-RS v2 endpoints.
     switch (wtParam) {
+      case "json":
```

Comment:
This feels like something that should be pluggable and stored in a Map instead of a switch. Thinking about what if somebody wants cbor or whatever.

Reply:
Yeah, I had this thought as I was adding the case.  A map would be a bit cleaner, especially as we add new content-types.

I'll leave it as-is for now, and switch it over the next time we add a content-type.